src/libstd/thread/mod.rs

```diff
@@ -1168,22 +1134,15 @@ impl Thread {
             EMPTY => return, // no one was waiting
             NOTIFIED => return, // already unparked
             PARKED => {} // gotta go wake someone up
-            _ => panic!("inconsistent state in unpark"),
+            n => panic!("inconsistent state in unpark: {}", n),
         }
 
-        // There is a period between when the parked thread sets `state` to
-        // `PARKED` (or last checked `state` in the case of a spurious wake
-        // up) and when it actually waits on `cvar`. If we were to notify
-        // during this period it would be ignored and then when the parked
-        // thread went to sleep it would never wake up. Fortunately, it has
-        // `lock` locked at this stage so we can acquire `lock` to wait until
-        // it is ready to receive the notification.
-        //
-        // Releasing `lock` before the call to `notify_one` means that when the
-        // parked thread wakes it doesn't get woken only to have to wait for us
-        // to release `lock`.
-        drop(self.inner.lock.lock());
-        self.inner.cvar.notify_one();
+        unsafe {
+            crate::parking_lot_core::unpark_all(
```

Comment:
Yes, but what would be the benefit of `unpark_one` then? The only difference I see is that it's more complex to call `unpark_one` since it requires a closure instead of just a hardcoded token.